stdlib/atomic.mli

```diff
@@ -17,6 +17,81 @@
 
 (** Atomic references.
 
+    A basic use case is to have global counters that are updated in a
+    thread-safe way:
```

Comment:
I like to have "real" examples, I suppose. What do the maintainers think? @Octachron @gasche for example

Reply:
I found this example a tad long as well, on the other hand the BFS sounded like a natural choice for queues. (I'm often too lazy to reach for a queue when I BFS and I use two lists, "frontier" and "next", but let's forget about this nitpick.) The Atomic example also had the issue of using Thread instead of Domain. It's fixed now (I haven't tried to run it yet, I'm in holidays until the end of the month and I somehow convinced myself that replying to messages is okay but running the compiler on PRs is not), but there is still a wide margin for debate. For example, the message we send is that `Domain` is the low-level, core module and that users should use higher-level abstractions (outside for the stdlib for now).

I think that the duty is on us to find a "real example" that is simpler but that @c-cube still likes, and maybe also manages to tiptoe around those difficulties.